lux-lib/src/operations/run_lua.rs

```diff
@@ -33,39 +38,68 @@ pub enum RunLuaError {
     },
     #[error(transparent)]
     Paths(#[from] PathsError),
+
+    #[error(transparent)]
+    Tree(#[from] TreeError),
 }
 
-pub async fn run_lua(
-    root: &Path,
-    tree: &Tree,
+#[derive(Builder)]
+#[builder(start_fn = new, finish_fn(name = run, vis = "pub"))]
+pub struct RunLuaBuilder<'a> {
+    root: &'a Path,
+    tree: &'a Tree,
+    config: &'a Config,
     lua_cmd: LuaBinary,
-    args: &Vec<String>,
-) -> Result<(), RunLuaError> {
-    let paths = Paths::new(tree)?;
-
-    let lua_cmd: PathBuf = lua_cmd.try_into()?;
-
-    let status = match Command::new(&lua_cmd)
-        .current_dir(root)
-        .args(args)
-        .env("PATH", paths.path_prepended().joined())
-        .env("LUA_PATH", paths.package_path().joined())
-        .env("LUA_CPATH", paths.package_cpath().joined())
-        .status()
-        .await
-    {
-        Ok(status) => Ok(status),
-        Err(err) => Err(RunLuaError::LuaCommandFailed {
-            lua_cmd: lua_cmd.to_string_lossy().to_string(),
-            source: err,
-        }),
-    }?;
-    if status.success() {
-        Ok(())
-    } else {
-        Err(RunLuaError::LuaCommandNonZeroExitCode {
-            lua_cmd: lua_cmd.to_string_lossy().to_string(),
-            exit_code: status.code(),
-        })
+    args: &'a Vec<String>,
+    prepend_test_paths: Option<bool>,
+    prepend_build_paths: Option<bool>,
+}
+
+impl RunLuaBuilder<'_> {
```

Comment:
todo: Please add a constraint on the `State`.

Something like this:

```suggestion
impl<State> RunLuaBuilder<'_, State>
where
    State: run_lua_builder::State + run_lua_builder::IsComplete,
```

See the other builders (in `operations::Sync` for example).
This ensures that the `run_lua` function can only be called when all required fields have been set.

Reply:
Where does run_lua_builder come from?